src/Agent/NewRelic/Agent/Core/Transformers/TransactionTransformer/ErrorEventMaker.cs

```diff
@@ -43,7 +55,40 @@ public ErrorEventWireModel GetErrorEvent(ImmutableTransaction immutableTransacti
             var priority = immutableTransaction.Priority;
 
             _attribDefs.GetTypeAttribute(TypeAttributeValue.TransactionError).TrySetDefault(attribValues);
+            SetErrorGroup(immutableTransaction.TransactionMetadata?.ReadOnlyTransactionErrorState?.ErrorData, attribValues);
             return new ErrorEventWireModel(attribValues, isSynthetics, priority);
         }
+
+        private IList<string> GetFormattedStackTrace(ErrorData errorData)
+        {
+            if (errorData.StackTrace == null)
+            {
+                return null;
+            }
+
+            var stackTrace = StackTraces.ScrubAndTruncate(errorData.StackTrace, _configurationService.Configuration.StackTraceMaximumFrames);
+            return stackTrace;
+        }
+
+        private void SetErrorGroup(ErrorData errorData, IAttributeValueCollection attribValues)
+        {
+            if (_configurationService.Configuration.ErrorGroupCallback == null)
+            {
+                return;
+            }
+
+            using (_agentTimerService.StartNew(SetErrorGroupSupportabilityName))
+            {
+                var callbackAttributes = attribValues.GetAllAttributeValuesDic();
+                if (errorData?.RawException != null)
+                {
+                    callbackAttributes.Add("exception", errorData.RawException);
+                }
+
+                callbackAttributes.Add("stack_trace", GetFormattedStackTrace(errorData));
```

Comment:
Maybe define a constant for the extra attributes being added?

Reply:
Done!